File: HSG_THANH_HOA_2/contestants/huynh_minh_thuan/BAI3.cpp
#include <bits/stdc++.h>
#define ll long long

using namespace std;

int main() {
    ifstream inp("BAI3.inp");
    ofstream out("BAI3.out");

    int n, k;
    inp >> n >> k;

    ll a[n], ans = 0;
    vector<vector<ll>> ranges(k);

    for (int i = 0; i < n; i++) {
        inp >> a[i];
        ranges[i % k].push_back(a[i]);
    }

    for (auto range: ranges) {
        sort(range.begin(), range.end());

        ll mid = range[range.size() / 2];

        for (ll val: range) {
            ans += abs(val - mid);
        }
    }

    out << ans;

    inp.close();
    out.close();

    return 0;
}

File: HSG_HAI_DUONG_22_23/contestants/nguyen_the_khoi/bai3.py
with open ("bai3.inp","r") as f:
    n,m=map(int,f.readline().split())
    a=list(map(int,f.readline().split()))
    f.close()
n-=1
s=0
i=0
for j in range (m):
    if a[i]>=a[n]:
        s+=a[i]
        a.pop(i)
    else :
        s+=a[n]
        a.pop(n)
    n-=1
with open ("bai3.out","w") as f:
    f.write(str(s))
    f.close()